### PR TITLE
18 M3C cards

### PR DESCRIPTION
**Pull request by tool4ever**

### CODE REVIEW

File: forge-gui/res/cardsfolder/upcoming/cayth_famed_mechanist.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Name:Cayth, Famed Mechanist
+ManaCost:1 U R W
+Types:Legendary Creature Dwarf Artificer
+PT:3/3
+K:Fabricate:1
+S:Mode$ Continuous | Affected$ Creature.Other+nonToken+YouCtrl | AddKeyword$ Fabricate:1 | Description$ Other nontoken creatures you control have fabricate 1.
+A:AB$ Charm | Cost$ 2 T | Choices$ DBPopulate,DBProliferate | Defined$ You
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/cayth_famed_mechanist.txt:7]: invalid param

File: forge-gui/res/cardsfolder/upcoming/hideous_taskmaster.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+Types:Creature Eldrazi
+PT:7/2
+K:Devoid
+T:Mode$ SpellCast | ValidCard$ Card.Self | Execute$ TakeOneEach | TriggerDescription$ When you cast this spell, for each opponent, gain control of up to one target creature that player controls until end of turn. Untap those creatures. They gain trample, haste, and annihilator 1 until end of turn.
+SVar:TakeOneEach:DB$ GainControl | ValidTgts$ Creature.OppCtrl | TgtPrompt$ Choose target creature each opponent controls | TargetMin$ 0 | TargetMax$ OneEach | TargetsForEachPlayer$ True | LoseControl$ EOT | Untap$ True | AddKWs$ Trample & Haste & Annihilator:1
+SVar:OneEach:PlayerCountOpponents$Amount
+K:Trample
+K:Haste
+K:Annihilator:1
+SVar:PlayMain1
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/hideous_taskmaster.txt:12]: incomplete, does that work?

File: forge-gui/res/cardsfolder/upcoming/trenchpost.txt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+Name:Trenchpost
+ManaCost:no cost
+Types:Land Locus
+A:AB$ Mana | Cost$ T | Produced$ C | SpellDescription$ Add {C}.
+A:AB$ Mill | Cost$ 3 T | NumCards$ X | ValidTgts$ Player | TgtPrompt$ Choose a player | AILogic$ EndOfOppTurn | SpellDescription$ Target player mills a card for each Locus you control.
+SVar:X:Count$TypeOnBattlefield.Locus+YouCtrl
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/upcoming/trenchpost.txt:6]: does this count correctly or need proper valid syntax?

Comment (Simisays) [reply forge-gui/res/cardsfolder/upcoming/trenchpost.txt:6]: Nope doesn't work

File: forge-gui/res/cardsfolder/upcoming/angelic_aberration.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+K:Devoid
+K:Flying
+K:Vigilance
+T:Mode$ ChangesZone | Origin$ Any | Destination$ Battlefield | ValidCard$ Card.Self | Execute$ TrigSac | TriggerDescription$ When CARDNAME enters the battlefield, sacrifice any number of creatures each with base power or toughness 1 or less. Create that many 4/4 colorless Eldrazi Angel creature tokens with flying and vigilance.
+SVar:TrigSac:DB$ Sacrifice | Defined$ You | Amount$ SacX | SacValid$ Creature.basePowerLE1,Creature.baseToughnessLE1 | SacMessage$ any number of creatures each with base power or toughness 1 or less | RememberSacrificed$ True | Optional$ True | SubAbility$ DBToken
+SVar:DBToken:DB$ Token | TokenAmount$ TokenX | TokenScript$ c_4_4_eldrazi_angel_flying_vigilance | TokenOwner$ You | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:SacX:Count$Valid Creature.basePowerLE1+YouCtrl,Creature.baseToughnessLE1+YouCtrl
+SVar:TokenX:Remembered$Amount
+Oracle:Devoid\nFlying, vigilance\nWhen Angelic Aberration enters the battlefield, sacrifice any number of creatures each with base power or toughness 1 or less. Create that many 4/4 colorless Eldrazi Angel creature tokens with flying and vigilance.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/angelic_aberration.txt:14]: DeckHas:Ability$Sacrifice|Token

File: forge-gui/res/cardsfolder/upcoming/cayth_famed_mechanist.txt [hunk 5]
```diff
@@ -0,0 +1,11 @@
+Name:Cayth, Famed Mechanist
+ManaCost:1 U R W
+Types:Legendary Creature Dwarf Artificer
+PT:3/3
+K:Fabricate:1
+S:Mode$ Continuous | Affected$ Creature.Other+nonToken+YouCtrl | AddKeyword$ Fabricate:1 | Description$ Other nontoken creatures you control have fabricate 1.
+A:AB$ Charm | Cost$ 2 T | Choices$ DBPopulate,DBProliferate | CharmNum$ 1
+SVar:DBPopulate:DB$ CopyPermanent | Choices$ Creature.token+YouCtrl | NumCopies$ 1 | Populate$ True | SpellDescription$ Populate.
+SVar:DBProliferate:DB$ Proliferate | SpellDescription$ Proliferate.
+DeckHas:Ability$Counters|Token|Proliferate
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/cayth_famed_mechanist.txt:10]: maybe add `DeckHints:Ability$Token`

File: forge-gui/res/cardsfolder/upcoming/chittering_dispatcher.txt [hunk 6]
```diff
@@ -0,0 +1,9 @@
+Name:Chittering Dispatcher
+ManaCost:2 G
+Types:Creature Eldrazi Drone
+PT:2/3
+K:Devoid
+K:Myriad
+T:Mode$ ChangesZone | Origin$ Battlefield | Destination$ Any | ValidCard$ Card.Self | Execute$ TrigToken | TriggerDescription$ When CARDNAME leaves the battlefield, create a 0/1 colorless Eldrazi Spawn creature token with "Sacrifice this creature: Add {C}."
+SVar:TrigToken:DB$ Token | TokenAmount$ 1 | TokenScript$ c_0_1_eldrazi_spawn_sac | TokenOwner$ You
+Oracle:Devoid\nMyriad\nWhen Chittering Dispatcher leaves the battlefield, create a 0/1 colorless Eldrazi Spawn creature token with "Sacrifice this creature: Add {C}."
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/chittering_dispatcher.txt:9]: DeckHas:Ability$Token|Sacrifice & Type$Spawn

File: forge-gui/res/cardsfolder/upcoming/filigree_racer.txt [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+T:Mode$ Attacks | ValidCard$ Card.Self | Execute$ TrigTrigger | TriggerDescription$ Whenever CARDNAME attacks, you may pay {E}{E}. When you do, target instant or sorcery card in your graveyard gains jump-start until end of turn. (You may cast that card from your graveyard by discarding a card in addition to paying its other costs. Then exile it.)
+SVar:TrigTrigger:AB$ ImmediateTrigger | Cost$ PayEnergy<2> | Execute$ DBPump | SpellDescription$ When you do, target instant or sorcery card in your graveyard gains jump-start until end of turn. (You may cast that card from your graveyard by discarding a card in addition to paying its other costs. Then exile it.)
+SVar:DBPump:DB$Pump | ValidTgts$ Instant.YouOwn,Sorcery.YouOwn | KW$ Jump-start | TgtZone$ Graveyard | PumpZone$ Graveyard
+K:Crew:1
+Oracle:When Filigree Racer enters the battlefield, you get {E}{E}{E}{E} (four energy counters).\nWhenever Filigree Racer attacks, you may pay {E}{E}. When you do, target instant or sorcery card in your graveyard gains jump-start until end of turn. (You may cast that card from your graveyard by discarding a card in addition to paying its other costs. Then exile it.)\nCrew 1
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/filigree_racer.txt:11]: SVar:HasAttackEffect:TRUE
DeckHints:Type$Instant|Sorcery
DeckHas:Ability$Graveyard|Discard

File: forge-gui/res/cardsfolder/upcoming/horizon_of_progress.txt [hunk 8]
```diff
@@ -0,0 +1,7 @@
+Name:Horizon of Progress
+ManaCost:no cost
+Types:Land
+A:AB$ ManaReflected | Cost$ T PayLife<1> | ColorOrType$ Type | Valid$ Land.YouCtrl | ReflectProperty$ Produce | SpellDescription$ Add one mana of any type that a land you control could produce.
+A:AB$ ChangeZone | Cost$ 3 T | Origin$ Hand | Destination$ Battlefield | Tapped$ True | ChangeType$ Land | ChangeNum$ 1 | Optional$ True | SpellDescription$ You may put a land card from your hand onto the battlefield tapped.
+A:AB$ Draw | Cost$ 1 T Sac<1/CARDNAME> | NumCards$ 1 | SpellDescription$ Draw a card.
+Oracle:{T}, Pay 1 life: Add one mana of any type that a land you control could produce.\n{3}, {T}: You may put a land card from your hand onto the battlefield tapped.\n{1}, {T}: Sacrifice Horizon of Progress: Draw a card.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/horizon_of_progress.txt:7]: DeckHas:Ability$Sacrifice

File: forge-gui/res/cardsfolder/upcoming/hourglass_of_the_lost.txt [hunk 9]
```diff
@@ -0,0 +1,8 @@
+Name:Hourglass of the Lost
+ManaCost:2 W
+Types:Artifact
+A:AB$ Mana | Cost$ T | Produced$ W | SubAbility$ DBPutCounter | SpellDescription$ Add {W}. Put a time counter on CARDNAME.
+SVar:DBPutCounter:DB$ PutCounter | CounterType$ TIME | CounterNum$ 1
+A:AB$ ChangeZoneAll | Cost$ T SubCounter<X/TIME> Exile<1/CARDNAME> | ChangeType$ Permanent.YouOwn+nonLand+cmcEQX | Origin$ Graveyard | Destination$ Battlefield | SorcerySpeed$ True | SpellDescription$ Return each nonland permanent card with mana value X from your graveyard to the battlefield. Activate only as a sorcery.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/hourglass_of_the_lost.txt:6]: Hmm
image
+ `CostDesc$ {T}, Remove X time counters from CARDNAME and exile it:`

File: forge-gui/res/cardsfolder/upcoming/hourglass_of_the_lost.txt [hunk 10]
```diff
@@ -0,0 +1,8 @@
+Name:Hourglass of the Lost
+ManaCost:2 W
+Types:Artifact
+A:AB$ Mana | Cost$ T | Produced$ W | SubAbility$ DBPutCounter | SpellDescription$ Add {W}. Put a time counter on CARDNAME.
+SVar:DBPutCounter:DB$ PutCounter | CounterType$ TIME | CounterNum$ 1
+A:AB$ ChangeZoneAll | Cost$ T SubCounter<X/TIME> Exile<1/CARDNAME> | ChangeType$ Permanent.YouOwn+nonLand+cmcEQX | Origin$ Graveyard | Destination$ Battlefield | SorcerySpeed$ True | SpellDescription$ Return each nonland permanent card with mana value X from your graveyard to the battlefield. Activate only as a sorcery.
+SVar:X:Count$xPaid
+Oracle:{T}: Add {W}. Put a time counter on Hourglass of the Lost.\n{T}, Remove X time counters from Hourglass of the Lost and exile it: Return each nonland permanent card with mana value X from your graveyard to the battlefield. Activate only as a sorcery.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/hourglass_of_the_lost.txt:8]: DeckHas:Ability$Counters ?
DeckHints:Ability$Counters too?

File: forge-gui/res/cardsfolder/upcoming/inversion_behemoth.txt [hunk 11]
```diff
@@ -0,0 +1,8 @@
+Name:Inversion Behemoth
+ManaCost:2 C C
+Types:Creature Eldrazi
+PT:2/9
+T:Mode$ Phase | Phase$ BeginCombat | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigPump | TriggerDescription$ At the beginning of combat on your turn, switch the power and toughness of each of any number of target creatures until end of turn.
+SVar:TrigPump:DB$ Pump | ValidTgts$ Creature | TargetMin$ 0 | TargetMax$ X | TgtPrompt$ Select target creature | KW$ HIDDEN CARDNAME's power and toughness are switched
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/inversion_behemoth.txt:6]: IMO - `TgtPrompt$ Select any number of target creatures`

File: forge-gui/res/cardsfolder/upcoming/hourglass_of_the_lost.txt [hunk 12]
```diff
@@ -0,0 +1,9 @@
+Name:Hourglass of the Lost
+ManaCost:2 W
+Types:Artifact
+A:AB$ Mana | Cost$ T | Produced$ W | SubAbility$ DBPutCounter | SpellDescription$ Add {W}. Put a time counter on CARDNAME.
+SVar:DBPutCounter:DB$ PutCounter | CounterType$ TIME | CounterNum$ 1
+A:AB$ ChangeZoneAll | Cost$ T SubCounter<X/TIME> Exile<1/CARDNAME> | CostDesc$ {T}, Remove X time counters from CARDNAME and exile it: | ChangeType$ Permanent.YouOwn+nonLand+cmcEQX | Origin$ Graveyard | Destination$ Battlefield | SorcerySpeed$ True | SpellDescription$ Return each nonland permanent card with mana value X from your graveyard to the battlefield. Activate only as a sorcery.
+SVar:X:Count$xPaid
+DeckHas:Ability$Counters
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/hourglass_of_the_lost.txt:8]: `|Graveyard`

File: forge-gui/res/cardsfolder/upcoming/lazotep_quarry.txt [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+ManaCost:no cost
+Types:Land Desert
+A:AB$ Mana | Cost$ T | Produced$ C | SpellDescription$ Add {C}.
+A:AB$ Mana | Cost$ T Sac<1/Creature> | Produced$ Any | SpellDescription$ Add one mana of any color.
+A:AB$ ChangeZone | Cost$ X 2 T Sac<1/Desert> | RememberLKI$ True | Origin$ Graveyard | Destination$ Exile | ValidTgts$ Creature.YouOwn+cmcEQX | ChangeNum$ 1 | SubAbility$ DBCopy | AILogic$ TheScarabGod | SorcerySpeed$ True | SpellDescription$ Exile target creature card with mana value X from your graveyard. Create a token that's a copy of it, except it's a 4/4 black Zombie. Activate only as a sorcery.
+SVar:DBCopy:DB$ CopyPermanent | Defined$ Remembered | SetPower$ 4 | SetToughness$ 4 | SetColor$ Black | SetCreatureTypes$ Zombie | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:X:Count$xPaid
+DeckHas:Ability$Mana.Colorless
+DeckHints:Color$Colorless & Keyword$Devoid
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/lazotep_quarry.txt:11]: & Type$Desert ?

File: forge-gui/res/cardsfolder/upcoming/lazotep_quarry.txt [hunk 14]
```diff
@@ -0,0 +1,12 @@
+Name:Lazotep Quarry
+ManaCost:no cost
+Types:Land Desert
+A:AB$ Mana | Cost$ T | Produced$ C | SpellDescription$ Add {C}.
+A:AB$ Mana | Cost$ T Sac<1/Creature> | Produced$ Any | SpellDescription$ Add one mana of any color.
+A:AB$ ChangeZone | Cost$ X 2 T Sac<1/Desert> | RememberLKI$ True | Origin$ Graveyard | Destination$ Exile | ValidTgts$ Creature.YouOwn+cmcEQX | ChangeNum$ 1 | SubAbility$ DBCopy | AILogic$ TheScarabGod | SorcerySpeed$ True | SpellDescription$ Exile target creature card with mana value X from your graveyard. Create a token that's a copy of it, except it's a 4/4 black Zombie. Activate only as a sorcery.
+SVar:DBCopy:DB$ CopyPermanent | Defined$ Remembered | SetPower$ 4 | SetToughness$ 4 | SetColor$ Black | SetCreatureTypes$ Zombie | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+SVar:X:Count$xPaid
+DeckHas:Ability$Mana.Colorless
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/lazotep_quarry.txt:10]: |Sacrifice

File: forge-gui/res/cardsfolder/upcoming/sage_of_the_maze.txt [hunk 15]
```diff
@@ -0,0 +1,10 @@
+Name:Sage of the Maze
+ManaCost:2 G
+Types:Creature Elf Wizard
+PT:1/3
+A:AB$ Mana | Cost$ T | Produced$ Combo Any | Amount$ 2 | SpellDescription$ Add two mana in any combination of colors.
+A:AB$ Animate | Cost$ T | ValidTgts$ Land.YouCtrl | TgtPrompt$ Select target land you control | Power$ X | Toughness$ X | Types$ Citizen,Creature | Keywords$ Haste | SorcerySpeed$ True | SpellDescription$ Until end of turn, target land you control becomes an X/X Citizen creature with haste in addition to its other types, where X is twice the number of Gates you control. Activate only as a sorcery.
+A:AB$ Untap | Cost$ tapXType<1/Gate> | SpellDescription$ Untap CARDNAME.
+SVar:X:Count$Valid Gate.YouCtrl/Twice
+DeckHints:Type$Gate
+Oracle:{T}: Add two mana in any combination of colors.\n{T}: Until end of turn, target land you control becomes an X/X Citizen creature with haste in addition to its other types, where X is twice the number of Gates you control. Activate only as a sorcery.\nTap an untapped Gate you control: Untap Sage of the Maze.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/sage_of_the_maze.txt:10]: DeckHas:Type$Citizen

File: forge-gui/res/cardsfolder/upcoming/sphinx_of_the_revelation.txt [hunk 16]
```diff
@@ -0,0 +1,12 @@
...
+Types:Artifact Creature Sphinx
+PT:4/5
+K:Flying
+K:Lifelink
+T:Mode$ LifeGained | ValidPlayer$ You | TriggerZones$ Battlefield | Execute$ TrigEnergy | TriggerDescription$ Whenever you gain life, you get that many {E} (energy counters).
+SVar:TrigEnergy:DB$ PutCounter | Defined$ You | CounterType$ ENERGY | CounterNum$ LifeX
+A:AB$ Draw | Cost$ W U U T PayEnergy<X> | NumCards$ X | SpellDescription$ Draw X cards.
+SVar:LifeX:TriggerCount$LifeAmount
+SVar:X:Count$xPaid
+Oracle:Flying, lifelink\nWhenever you gain life, you get that many {E} (energy counters).\n{W}{U}{U}, {T}, Pay X {E}: Draw X cards.
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/sphinx_of_the_revelation.txt:12]: DeckHas:Ability$LifeGain
DeckHints:Ability$LifeGain

File: forge-gui/res/cardsfolder/upcoming/stone_idol_generator.txt [hunk 17]
```diff
@@ -0,0 +1,9 @@
+Name:Stone Idol Generator
+ManaCost:5
+Types:Artifact
+T:Mode$ Attacks | ValidCard$ Creature.YouCtrl | TriggerZones$ Battlefield | Execute$ TrigEnergy | TriggerDescription$ Whenever a creature you control attacks, you get {E} (an energy counter).
+SVar:TrigEnergy:DB$ PutCounter | Defined$ You | CounterType$ ENERGY | CounterNum$ 1
+A:AB$ Token | Cost$ T PayEnergy<6> | TokenScript$ c_6_12_a_construct_trample | SorcerySpeed$ True | SpellDescription$ Create a 6/12 colorless Construct artifact creature token with trample. Activate only as a sorcery.
+SVar:PlayMain1:TRUE
+DeckHints:Ability$Token
```
Comment (Northmoc) [inline forge-gui/res/cardsfolder/upcoming/stone_idol_generator.txt:8]: & Type$Construct

Comment (tool4ever) [reply forge-gui/res/cardsfolder/upcoming/stone_idol_generator.txt:8]: only makes sense as DeckHas?
I've kind of phased that out with my last PR

**Review by Northmoc — APPROVED** [review]
LGTM! 👍

**Pull request merged**